crates/provider-sdk/src/lib.rs

```diff
@@ -19,6 +16,12 @@ pub use provider::ProviderConnection;
 pub use provider_main::{load_host_data, run_provider, start_provider};
 pub use rpc_client::RpcClient;
 
+pub mod core {
+    pub use wasmcloud_core::{
+        HealthCheckRequest, HealthCheckResponse, HostData, LinkDefinition, WasmCloudEntity,
+    };
+}
```

Comment:
I would expect this to be removed and any usages of these to be replaced by imports from `wasmcloud_core`

Reply:
I was conflicted on this. Do you think it's better for providers to explicitly depend on `wasmcloud_core`, or re-export the types providers need so they only have to depend on `wasmcloud_provider_sdk`?

Reply:
I'd definitely re-export, but I would not do the inner nesting, i.e. I would do `pub use wasmcloud_core as core` and call it a day

Reply:
Fixed